onchain/src/quests/template_quest.cairo

```diff
@@ -0,0 +1,69 @@
+use art_peace::templates::interfaces::{
+    ITemplateStoreDispatcher, ITemplateStoreDispatcherTrait, ITemplateVerifierDispatcher,
+    ITemplateVerifierDispatcherTrait, TemplateMetadata
+};
+
+#[starknet::contract]
+pub mod TemplateQuest {
+    use starknet::{ContractAddress, get_caller_address};
+    use art_peace::{IArtPeaceDispatcher, IArtPeaceDispatcherTrait};
+    use art_peace::quests::{IQuest, QuestClaimed};
+
+    #[storage]
+    struct Storage {
+        art_peace: IArtPeaceDispatcher,
+        reward: u32,
+        claimed: LegacyMap<ContractAddress, bool>,
+    }
+
+
+    #[abi(embed_v0)]
+    impl TemplateQuest of IQuest<ContractState> {
+        fn get_reward(self: @ContractState) -> u32 {
+            self.reward.read()
+        }
+
+        fn is_claimable(
+            self: @ContractState, user: ContractAddress, calldata: Span<felt252>
+        ) -> bool {
+            let art_peace = self.art_peace.read();
+
+            if self.claimed.read(user) {
+                return false;
+            }
+
+            let template_store = ITemplateStoreDispatcher { 
```

Comment:
You can move this to a storage variable as mentioned in the other comment. And set it up in the constructor.

Reply:
Alright